File: src/javascript/user/message/forward-message.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

if (process.env.FAXCORE_CONFIRM_DESTRUCTIVE !== 'true') {
  throw new Error('Set FAXCORE_CONFIRM_DESTRUCTIVE=true to forward a message.');
}

const messageID = process.env.FAXCORE_MESSAGE_ID;
const usernames = process.env.FAXCORE_TARGET_USERNAMES;

if (!messageID || !usernames) {
  throw new Error('FAXCORE_MESSAGE_ID and FAXCORE_TARGET_USERNAMES are required.');
}

const client = new FaxCoreClient();
const result = await client.post('/api/message/forward', {
  messageID,
  usernames: usernames.split(',').map((value) => value.trim()).filter(Boolean)
});

printResult('Forward message', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/Message/ForwardMessage.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Message
{
    internal static class ForwardMessage
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            if (!string.Equals(Environment.GetEnvironmentVariable("FAXCORE_CONFIRM_DESTRUCTIVE"), "true", StringComparison.OrdinalIgnoreCase))
            {
                throw new InvalidOperationException("Set FAXCORE_CONFIRM_DESTRUCTIVE=true to forward a message.");
            }

            if (string.IsNullOrWhiteSpace(config.MessageId) || string.IsNullOrWhiteSpace(config.TargetUsernames))
            {
                throw new InvalidOperationException("Set FAXCORE_MESSAGE_ID and FAXCORE_TARGET_USERNAMES before running user.message.forward.");
            }

            return client.PostAsync("/api/message/forward", new
            {
                messageID = config.MessageId,
                usernames = config.TargetUsernames.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries)
            });
        }
    }
}
